### ISSUE TITLE
Helm3 : helm rollback with multiple replicas does not work as expected

### ISSUE THREAD

**Issue by pragyamehta** [helm/helm#5951]
Repro steps:
1. helm install chart
Chart has only one deployment file : 
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: pingapp
  labels:
    app: pingapp
spec:
  replicas: 2
  selector:
    matchLabels:
      app: pingapp
  template:
    metadata:
      labels:
        app: pingapp
    spec:
      containers:
      - name: pingapp
        image: hello-world
        ports:
        - containerPort: 80
```

2 pods will get created, both with crashloopbackoff

2. Update the deployment file's image name to a working image e.g. murph15/pingapp

3. helm upgrade the release

4. 2 pods are now running successfully with the new image

5. Now, helm rollback to revision 1

6. **Expectation** : 2 pods both with image hello-world should exist
**Reality**: 3 pods exist - 2 with hello-world image and 1 with murph15/pingapp image.

Output of `helm version`:
`version.BuildInfo{Version:"v3.0.0-alpha.1", GitCommit:"b9a54967f838723fe241172a6b94d18caf8bcdca", GitTreeState:"clean"}`

Output of `kubectl version`:
`Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.5", GitCommit:"51dd616cdd25d6ee22c83a858773b607328a18ec", GitTreeState:"clean", BuildDate:"2019-01-16T18:14:49Z", GoVersion:"go1.10.7", Compiler:"gc", Platform:"linux/amd64"}`

Cloud Provider/Platform (AKS, GKE, Minikube etc.): AKS

**Comment by hickeyma** [comment]
I tested this with a scaffold chart and it works using replica set changes from 2 to 3 and rollback. This when using the same image.

@pragyamehta Does it work as expected when the image is the same for the replicas?

**Comment by pragyamehta** [comment]
Yes, it works when using the same image with varying pod instances. It fails when I change the image.
You can try this scenario using the same image names as I have mentioned with the same repro steps. The images are public images. (Please note, I do not change the replica count during the repro, it is always 2)

**Comment by bacongobbler** [comment]
I'm going to close this ticket due to inactivity, but please re-open if this still needs to be addressed. I'm fairly certain this workflow has been fixed in Helm 3 with the new three-way strategic merge patch logic. Thanks!

**Comment by martin-schulze-e2m** [comment]
I am not sure if this is the same issue but it sounds similar: We rolled out an update to one of our charts which referenced a non existing docker tag. Once the error was noticed, we tried a rollback. We got a working deployment of the old image but also one leftover pending pod with the new, non-existing image. I noticed that I could not even scale the responsible ReplicaSet to zero. Only `kubectl rollout undo --to-revision ...` worked.
